airflow/providers/amazon/aws/hooks/rds.py

```diff
@@ -240,7 +240,7 @@ def get_db_instance_state(self, db_instance_id: str) -> str:
         try:
             response = self.conn.describe_db_instances(DBInstanceIdentifier=db_instance_id)
         except self.conn.exceptions.ClientError as e:
-            if e.response["Error"]["Code"] == "DBInstanceNotFoundFault":
+            if e.response["Error"]["Code"] == "DBInstanceNotFound":
```

Comment:
I wonder if we need to check both values since no one reported the issues before.
```suggestion
            if e.response["Error"]["Code"] in ["DBInstanceNotFoundFault", "DBInstanceNotFound"]:
```
However if @Taragolis solution is applicable, it would be much better.